core/src/main/java/org/opensearch/sql/analysis/Analyzer.java

```diff
@@ -210,6 +204,21 @@ public LogicalPlan visitTableFunction(TableFunction node, AnalysisContext contex
         tableFunctionImplementation.applyArguments());
   }
 
+  // This method has optimization to avoid metadata search calls to OpenSearch
+  // by checking if the name has only one part in it.
+  private Set<String> getAllowedDataSources(QualifiedName qualifiedName) {
```

Comment:
this function expose DataSourceService internal logic to external. i think 
1. we need another api on datasourceservice to retrieve allowedDataSourceNames based on qulifiedName.
2. or we need api to resolve DataSourceSchemaIdentifierNameResolver.

Reply:
Will move this logic to DSS or Resolver